api/provider/aws/test_aws/environment_scaler_test.go

```diff
@@ -0,0 +1,351 @@
+package test_aws
+
+import (
+	"strconv"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/golang/mock/gomock"
+	"github.com/quintilesims/layer0/api/job/mock_job"
+	provider "github.com/quintilesims/layer0/api/provider/aws"
+	"github.com/quintilesims/layer0/api/provider/mock_provider"
+	awsc "github.com/quintilesims/layer0/common/aws"
+	"github.com/quintilesims/layer0/common/config/mock_config"
+
+	"github.com/quintilesims/layer0/common/models"
+	bytesize "github.com/zpatrick/go-bytesize"
+)
+
+type EnvironmentScalerUnitTest struct {
+	ExpectedScale     int
+	MemoryPerProvider bytesize.Bytesize
+	ResourceProviders []*models.Resource
+	ResourceConsumers []models.Resource
+}
+
+// Add Old Test Verbatim from Dev PR
+
+func (e *EnvironmentScalerUnitTest) Run(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
```

Comment:
this all needs to be split out - we should be testing the scaling algorithm as a single unit